_posts/2019-03-14-2019-kafka-connector.md

```diff
@@ -0,0 +1,183 @@
+
+## Deploy the components
+
+In order to deploy the whole stack and play around with it go ahead and apply the kubernetes files while inside the kafka-connector folder:
+
+```bash
+$ kubectl apply -f yaml/kubernetes
+```
+
+We will deploy simple producer provided by Alex:
+
+```bash
+$ kubectl run -n openfaas -t -i --image alexellis2/kafka-producer:0.2.0 producer
+```
+
+Congrats now you have the whole stack deployed:
+* Zookeeper
+* Producer Service
+* Broker Host
+* Kafka Connector
+
+Before we proceed, make sure you have all the components running:
+
+```bash
+$ kubectl get pods -n openfaas
+NAME                               READY   STATUS    RESTARTS   AGE
+alertmanager-b55c578d4-87dgf       1/1     Running   0          37m
+faas-idler-7945c9bc7b-74z68        1/1     Running   1          37m
+gateway-7b7455dfd8-kzktf           2/2     Running   1          37m
+kafka-broker-544fbccd48-b5f65      1/1     Running   0          33m
+kafka-connector-7898548745-5wclq   1/1     Running   0          33m
+nats-6686bb4b95-lj6b8              1/1     Running   0          37m
+producer-c77fb7ccf-cx8xg           1/1     Running   0          31m
+prometheus-7dd5d65568-bkr8j        1/1     Running   0          37m
+queue-worker-8655bb54cb-q5nk8      1/1     Running   1          37m
+zookeeper-699f568f6f-wpzb6         1/1     Running   0          33m
+
+```
+
+## Subscribe function
+
+In order to consume topics from the connector we should just apply annotation with key `topic` and value whichever you used in the `connector-dep.yml` in our case it was `faas-request`.
+
+We will use function from the store called `figlet`:
+
+```bash
+$ faas-cli store deploy figlet --annotation "topic=faas-request"
+```
+
+Voila, your function is now subscribed to the topic and it can be invoked by producing message on that topic.
+
+Now you might ask:
+
+"Ok, cool but how am I supposed see the output of my function?"
+
+Remember in the configuration we set `print_response_body` to `true` so we can see the response body in the logs of the Kafka Connector.
+
+## Generate load on a topic
+
+Lets proceed by opening two terminals, one will be used to check the output of the function and the other will produce load on the topic.
+
+In the first terminal open shell session inside producer pod:
+
+```bash
+$ kubectl exec -it $(kubectl get pods -o name -n openfaas | grep -m1 producer | cut -d'/' -f 2) -n openfaas /bin/sh
+```
+
+In the second terminal follow the Kafka Connector logs with this command:
+
+```bash
+$ kubectl logs $(kubectl get pods -o name -n openfaas | grep -m1 kafka-connector | cut -d'/' -f 2) -n openfaas --tail 100 -f
+```
+
+In the first terminal, inside the producer pod shell session invoke the producer binary with proper command line arguments:
+
+```bash
+/usr/bin producer -messages=25 -broker=kafka:9092 -pause 1s -topic=faas-request
+```
+
+You should see the output of the function in the kafka-connector logs.
+
+## Wrapping up
```

Comment:
I think we need a bit more here to stimulate people's thinking.  A real world use case perhaps?  Online order triggering billing , invoicing, and warehousing functions?

Reply:
I will spend some time how to formulate those, good proposal 👍

Reply:
Added paragraph before that one showing the real life example 💯